### PR TITLE
(feat) Update diagnosis to support `rank` and `certainty`

### PR DESCRIPTION
**Pull request by donaldkibet**
# Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary

This PR adds support to create diagnosis by specifying `rank` `certainty` coded diagnosis`. In a follow up PR will add support for non-coded diagnosis.

## Screenshots
Kapture 2023-09-20 at 00 24 44

### CODE REVIEW

**Comment by ibacher** [comment]
How does this interact with the [previous implementation of this behavior](https://github.com/openmrs/openmrs-ngx-formentry/pull/37) (understanding that this does not really support diagnostic certainty). In particular, I'm concerned with how rank is handled, because it's not uncommon to have a requirement that is something like "The chief complaint should be listed first on the form", which helps if rank is an attribute of the question rather than something user-determined.

Basically, as long as we can go with both options, I think it's fine.

**Comment by icrc-agomes** [comment]
Hello @donaldkibet on our distribution we are using diagnoses. 

We updated our configuration to work as intended in this PR, everything went well with the creation of diagnoses.

However, when editing and eliminating any diagnoses the payload that was sent to the server went with all values voided equal to false.

image


Has you can see in this image I had 2 diagnoses and I deleted both of them, yet it still send both elements with property voided as false.


Configuration used:

```
{
  "label": "XXX",
  "sections": [
    {
      "label": "XXX",
      "isExpanded": "true",
      "questions": [
        {
          "isExpanded": "true",
          "questions": [
            {
              "label": "Diagnoses",
              "type": "diagnosisGroup",
              "questionOptions": {
                "rendering": "repeating"
              },
              "questions": [
                {
                  "label": "Diagnosis",
                  "id": "primaryDiagnosisId",
                  "type": "diagnosis",
                  "questionOptions": {
                    "rendering": "diagnosis",
                    "diagnosisType": "diagnosis"
                  },
                  "validators": []
                },
                {
                  "id": "DiagnosisCertainty",
                  "label": "Diagnosis Certainty",
                  "type": "diagnosis",
                  "questionOptions": {
                    "rendering": "radio",
                    "diagnosisType": "certainty",
                    "answers": [
                      {
                        "concept": "CONFIRMED",
                        "label": "Confirmed"
                      },
                      {
                        "concept": "PROVISIONAL",
                        "label": "Provisional"
                      }
                    ]
                  },
                  "validators": []
                },
                {
                  "id": "DiagnosisRank",
                  "label": "Diagnosis Rank",
                  "type": "diagnosis",
                  "questionOptions": {
                    "rendering": "radio",
                    "diagnosisType": "rank",
                    "answers": [
                      {
                        "concept": 1,
                        "label": "Primary"
                      },
                      {
                        "concept": 2,
                        "label": "Secondary"
                      }
                    ]
                  },
                  "validators": []
                }
              ],
              "hide": {
                "hideWhenExpression": "patientReceivedOrReceivingTreatment != '1065AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' && patientReceivedOrReceivingTreatment != '671ebfb8-bccd-479a-ad96-5b2b6971688a' && patientReceivedOrReceivingTreatment != '5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA'"
              }
            }
          ]
        }
      ]
    }
  ]
}

```

**Comment by donaldkibet** [comment]
@icrc-agomes, I've made the requested changes. @ibacher, I've retained the distinction between primary and secondary diagnoses as per the existing work.

**Review by ibacher — APPROVED** [review]
